### PR TITLE
mitmweb: Make columns configurable/customizable

### PR DESCRIPTION
**Pull request by gorogoroumaru**
#### Description

issue #4107 and issue #4092 

I made columns in mitmweb configurable and customizable using option.

### CODE REVIEW

File: mitmproxy/tools/web/webaddons.py [hunk 1]
```diff
@@ -21,6 +22,10 @@ def load(self, loader):
             "web_host", str, "127.0.0.1",
             "Web UI host."
         )
+        loader.add_option(
+            "columns", Sequence[str], ["", "Path", "Method", "Status", "Size", "Time"],
```
Comment (mhils) [inline mitmproxy/tools/web/webaddons.py:26]: Can we call this web_columns (or web_flowlist_columns?) please to make clear that this is about mitmweb only for now? Also let's make the help text a bit more descriptive. How about "Columns to show in the flow list."?

File: web/src/css/flowtable.less [hunk 2]
```diff
@@ -114,16 +114,19 @@
         }
     }
     .col-method {
-        width: 60px;
+        width: auto;
```
Comment (mhils) [inline web/src/css/flowtable.less:117]: What's the motivation for adjusting the width here? IMO this looks less desirable on certain screen sizes now.

File: web/src/js/__tests__/ducks/tutils.js [hunk 3]
```diff
@@ -77,6 +77,13 @@ export function TStore(){
                 type: "str",
                 value: "b"
             },
+            columns: {
+                choices: null,
+                default: ["Path"],
+                help: "Display specified columns",
+                type: "sequence of str",
+                value: ["", "Path", "Method", "Status", "Size", "Time"]
```
Comment (mhils) [inline web/src/js/__tests__/ducks/tutils.js:85]: The empty string for the TLS indicator is really awful here. Can we please use some other identifer? How about `column.name.slice(0, -6).toLowerCase()`? Alternative, we can add an explicit one.

File: web/src/js/ducks/options.js [hunk 4]
```diff
@@ -6,7 +6,11 @@ export const RECEIVE = 'OPTIONS_RECEIVE'
 export const UPDATE = 'OPTIONS_UPDATE'
 export const REQUEST_UPDATE = 'REQUEST_UPDATE'
 
-const defaultState = {}
+const defaultState = {
+    columns: {
+        value: ["", "Path", "Method", "Status", "Size", "Time"]
+    }
```
Comment (mhils) [inline web/src/js/ducks/options.js:12]: Let's keep this empty here (and just have the FlowRow fall back to defaults if the option is not set)

**Review by mhils — COMMENTED** [review]
Thanks! This looks very nice 😃 
I have a few smaller comments on the code, but overall I'd be very happy to get this in. 😊

**Comment by gorogoroumaru** [comment]
Sorry, I lost my local repository.
I will create another PR.

**Pull request closed without merging**